server/resolve.js

```diff
@@ -51,5 +52,33 @@ async function isFile (p) {
     if (err.code === 'ENOENT') return false
     throw err
   }
+
+  // We need the path to be case sensitive
+  const realpath = await getTrueFilePath(p)
+  if (p !== realpath) return false
```

Comment:
Currently, this is in a utility library and passing `dev` option need to change in few places. And those changes seems to be looks nice. If we are using NODE_ENV then it would have been simple.

Here even in development, when clicked on link with an incorrect case, user will get a 404. I think that's a good way to show the problem.